src/platform/windows.rs

```diff
@@ -117,6 +117,12 @@ pub trait WindowExtWindows {
   /// Whether to show the window icon in the taskbar or not.
   fn set_skip_taskbar(&self, skip: bool);
 
+  /// Sets the Taskbar Progress State
+  fn set_taskbar_progress_state(&self, state: i32);
+
+  /// Sets the Taskbar Progress
+  fn set_taskbar_progress(&self, current: u64, total: u64);
+
```

Comment:
lets combine these two methods into one method and move it back to `src/window.rs`:
```suggestion
  fn set_progress_bar(&self, state: ProgressBarState);
```
and

```
pub enum ProgressState{
  None,
  Normal,
  Intermediate,
  Paused,
}

pub struct ProgressBarState {
	pub state: ProgressState,
	pub total: u32,
	pub current: u32,
	/// **Linux Only**
	pub unity_uri: Option<String>,
}
```

Reply:
I think keeping them separate is better? Because You can change the progress without changing the state (can save in computation?).

Reply:
Also, It'll affect the windows version too

Reply:
> Because You can change the progress without changing the state

the `state` field could be an `Option<ProgressState>` if it is `None`, we keep the old progress state and update only the progress value.

Actually, lets also remove `total` field and change `current` to `progress` and document that progress should be between `0` and `100` and have the total internally fixed to `100` so it would be like this
```rs
pub struct ProgressBarState {
	pub state: ProgressState,
	pub progress: u32,
	/// **Linux Only**
	pub unity_uri: Option<String>,
}
```

> Also, It'll affect the windows version too

That's fine

Reply:
✅

Reply:
Shall Implement after I triage the zbus impl not working

Reply:
I think let's keep the total, that makes it better

Reply:
I don't see a use-case for `total` field tbh, we can just assume the total to be `100` and `current` should be between `0` and `100`

Reply:
Actually, I have a use case
Like the user is implementing a file downloader, then it'll it easier for supplying `current, total` instead of converting that into `current * 100 / total` for that person (considering he's starting at rust and doesn't know a lot)

Reply:
I'd rather keep the calculation on the user side of things instead of in the library code. There is use-cases when the total is not only downloading the file and might be related to installing it as well. In any use-case, the calculation is super simple that I don't see any value adding it inside the library.

Reply:
That works, though there will be an internal counter that converts anything > 100 as 100